### PR TITLE
[trainer] feat: Support RL trainer with TorchtitanEngine

### PR DESCRIPTION
**Pull request by acisseJZhong**
### What does this PR do?

Add RL support for TorchtitanEngine. Tested FSDP, TP/SP on Qwen3 0.6b and verified parity with FSDP engine.  

Note: 
- TP requires Torchtitan to have full DTensor: https://github.com/pytorch/torchtitan/pull/2149 (will land soon)
- Will add PP and EP support, and set up integration/CI test for torchtitan engine as follow up PRs.
- Torchtitan critic model is not supported, will follow up. GRPO doesn't use critic. 

### Test
```
GRPC_ENABLE_FORK_SUPPORT=0 NCCL_NVLS_ENABLE=0 MODEL_ID=Qwen/Qwen3-0.6B bash tests/special_e2e/run_ppo_trainer_torchtitan.sh

NUM_GPUS=4 FSDP_SIZE=2 TP_SIZE=2 GRPC_ENABLE_FORK_SUPPORT=0 NCCL_NVLS_ENABLE=0  bash tests/special_e2e/run_ppo_trainer_torchtitan.sh
```
[Package versions I am using](https://gist.github.com/acisseJZhong/e6f3920df5c5d88fef86750ce8518141), [version 2](https://gist.github.com/acisseJZhong/6cf2c14eab40a5c2a129443027894317)
[Torchtitan engine config](https://gist.github.com/acisseJZhong/6e68366c50b8efa396bcb50da7701a84)
[FSDP engine config](https://gist.github.com/acisseJZhong/9decbe8a22c3f5b0499f42f21229e10d)

test on qwen3 0.6b single process
- **critic scores, actor pg loss, actor loss are matching, and Titan step time is significantly less than FSDP engine**

image
image
image

test on qwen3 0.6b fsdp_size=2, tp_size=2
image
image
image

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

**Comment by wuxibin89** [comment]
> test on qwen3 0.6b single process, critic scores, actor pg loss, actor loss are matching, and Titan step time is significantly less than FSDP engine

@acisseJZhong I'm wondering where's the performance gain in FSDP vs Titan?

File: verl/trainer/config/engine/torchtitan.yaml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # Target class for this configuration
 _target_: verl.workers.config.TorchtitanEngineConfig
 
+# Whether to offload model parameters to CPU
+param_offload: False
```
Comment (wuxibin89) [inline verl/trainer/config/engine/torchtitan.yaml:5]: Please test when `param_offload=True` and `optimizer_offload=True`.

**Comment by acisseJZhong** [comment]
> where's the performance gain in FSDP vs Titan?

Have not tested extensively, the performance difference between Titan FSDP and FSDP engine might be from:
1. FSDP engine uses HF model with flash attention path, Torchtitan Engine uses its own model definition with flex/varlen attention
2. difference in handling variable length sequence: FSDP engine reshapes to (1, total_nnz) and relies on HF's flash attention integration with attention_mask=None; Torchtitan Engine constructs proper attention masks instead
3. difference in applying fspd v2: Torchtitan engine apply fsdp [here](https://github.com/pytorch/torchtitan/blob/5580c097b510814ebf6f9ef6200edbfa42199fb6/torchtitan/models/llama4/parallelize.py#L343), which looks different than fsdp engine's apply fsdp. 

As a follow up, I will probably benchmark larger models with parallelism between Torchtitan, FSDP, and Megatron.

File: verl/trainer/config/engine/torchtitan.yaml (hunk 1 above)
Comment (acisseJZhong) [reply verl/trainer/config/engine/torchtitan.yaml:5]: image
image
image


(also tested fsdp engine with FSPDv2, seems slightly faster than v1)
